http/src/request/multipart.rs

```diff
@@ -111,49 +41,117 @@ impl Default for Form {
         }
 
         Self {
+            attachments: &[],
+            boundary,
+            payload_json,
+        }
+    }
+
+    pub const fn attachments(mut self, attachments: &'a [AttachmentFile<'a>]) -> Self {
+        self.attachments = attachments;
+
+        self
+    }
+
+    pub fn build(self) -> Form {
+        let FormBuilder {
+            attachments,
             boundary,
-            buffer: Vec::new(),
+            payload_json,
+        } = self;
+
+        let mut buffer = Vec::new();
```

Comment:
The capacity of the buffer could be calculated here exactly, that would save some reallocations for the vec.